pkg/controller/sparkapplication/controller.go

```diff
@@ -597,6 +597,11 @@ func (c *Controller) syncSparkApplication(key string) error {
 			glog.Errorf("failed to update SparkApplication %s/%s: %v", app.Namespace, app.Name, err)
 			return err
 		}
+
+		if err := c.cleanUpBatchScheduling(app, appCopy); err != nil {
```

Comment:
Just realized that we have a method for cleaning up application resources like the driver UI service. We should add deletion of this resource to that method.

Reply:
Thanks for review!
I wonder if the function you are talking about is [deleteSparkResources](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/pkg/controller/sparkapplication/controller.go#L804).
As far as I understand, this function is called before the Spark application retry, which is called when not normal (invalidating, failed submission, etc.).
In the case of cleanup, the podgroup is deleted when it is in completed and failed state. This means that the spark application no longer restarts. If spark operator deletes the podgroup in the state of invalidating or failed submission, the spark application will not be queued to the volcano and will not be restarted until re-queued.
Please comment if it is different from what I understood.

Reply:
Makes sense.